src/barretenberg_structures.rs

```diff
@@ -1313,7 +1313,11 @@ impl TryFrom<&Circuit> for ConstraintSystem {
                 Opcode::Directive(_) | Opcode::Brillig(_) => {
                     // Directives, Oracles and Brillig are only needed by the pwg
                 }
-                Opcode::MemoryOp { block_id, op } => {
+                Opcode::MemoryOp {
+                    block_id,
+                    op,
+                    predicate: _,
```

Comment:
For now I have not done anything with the predicate, but this feels like it needs to be threaded into the backend or the predicates which have a value of zero need to be removed from the final ACIR by the ACVM.